src/Symfony/Component/Security/Http/Authenticator/AccessToken/FormEncodedBodyExtractor.php

```diff
@@ -0,0 +1,43 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Security\Http\Authenticator\AccessToken;
+
+use Symfony\Component\HttpFoundation\Request;
+
+/**
+ * Extracts a token from the body request.
+ *
+ * @author Florent Morselli <florent.morselli@spomky-labs.com>
+ *
+ * @see https://datatracker.ietf.org/doc/html/rfc6750#section-2.2
+ */
+final class FormEncodedBodyExtractor implements AccessTokenExtractorInterface
+{
+    public const DEFAULT_PARAMETER = 'access_token';
+
+    public function __construct(private readonly string $parameter = self::DEFAULT_PARAMETER)
+    {
+    }
+
+    public function extractAccessToken(Request $request): ?string
+    {
+        if (Request::METHOD_POST !== $request->getMethod()) {
+            return null;
+        }
+        if (!str_starts_with($request->headers->get('CONTENT_TYPE', ''), 'application/x-www-form-urlencoded')) {
```

Comment:
```suggestion
        if ('form' === $request->getContentType()) {
```

Reply:
I committed the suggestion but reverted it.
Actually, only `application/x-www-form-urlencoded` is mentioned in the RFC and `getContentType` returns `form` for this type and `multipart/form-data`.